core/src/main/java/discord4j/core/object/entity/channel/AllowedMentions.java

```diff
@@ -0,0 +1,222 @@
+
+    /**
+     * Sets the default types which are parsed in allowed mentions
+     * @param types the types to allow by default
+     */
+    public static void setDefaultParsedTypes(Possible<List<AllowedMentions.Type>> types) {
+        DEFAULT_TYPES = types;
+    }
+
+    /**
+     * Sets the default types which are parsed in allowed mentions
+     * @param types the types to allow by default
+     */
+    public static void setDefaultParsedTypes(List<AllowedMentions.Type> types) {
+        setDefaultParsedTypes(Possible.of(types));
+    }
+
+    /**
+     * Sets the default users which are allowed in mentions
+     * @param users the users to allow by default
+     */
+    public static void setDefaultAllowedUsers(Possible<List<Snowflake>> users) {
+        DEFAULT_USERS = users;
+    }
+
+    /**
+     * Sets the default users which are allowed in mentions
+     * @param users the users to allow by default
+     */
+    public static void setDefaultAllowedUsers(List<Snowflake> users) {
+        setDefaultAllowedUsers(Possible.of(users));
+    }
+
+    /**
+     * Sets the default roles which are allowed in mentions
+     * @param roles the roles to allow by default
+     */
+    public static void setDefaultAllowedRoles(Possible<List<Snowflake>> roles) {
+        DEFAULT_ROLES = roles;
+    }
+
+    /**
+     * Sets the default roles which are allowed in mentions
+     * @param roles the roles to allow by default
+     */
+    public static void setDefaultAllowedRoles(List<Snowflake> roles) {
+        setDefaultAllowedRoles(Possible.of(roles));
+    }
+
+    private final Possible<List<AllowedMentions.Type>> parse;
+    private final Possible<List<Snowflake>> userIds;
+    private final Possible<List<Snowflake>> roleIds;
+
+    private AllowedMentions(final Possible<List<AllowedMentions.Type>> parse, final Possible<List<Snowflake>> userIds,
+                            final Possible<List<Snowflake>> roleIds) {
+        this.parse = parse;
+        this.userIds = userIds;
+        this.roleIds = roleIds;
+    }
+
+    private <T, U> List<T> mapList(final List<U> list, final Function<? super U, ? extends T> mapper) {
+        final List<T> data = new ArrayList<>(list.size());
+        list.forEach(u -> data.add(mapper.apply(u)));
+        return data;
+    }
+
+    /**
+     * Maps this {@link AllowedMentions} object to a {@link AllowedMentionsData} JSON
+     * @return JSON object
+     */
+    public AllowedMentionsData toData() {
+        final ImmutableAllowedMentionsData.Builder builder = AllowedMentionsData.builder();
+        if (!parse.isAbsent()) {
+            builder.parse(mapList(parse.get(), Type::getRaw));
+        }
+        if (!userIds.isAbsent()) {
+            builder.users(mapList(userIds.get(), Snowflake::asString));
+        }
+        if (!roleIds.isAbsent()) {
+            builder.roles(mapList(roleIds.get(), Snowflake::asString));
+        }
+        return builder.build();
+    }
+
+    public static class Builder {
```

Comment:
If we know for certain we're moving from specs to builders eventually, I'm fine with this being a builder already. Otherwise, it should be a spec so we can migrate everything in the same way.

Reply:
As it gets created with a static call `AllowedMentions.builder()` I believe that "builder" is the correct naming here.
When seeing Discord4J specs, specs are mostly used in a consumer context and I'd try to avoid to have a consumer inside of a consumer just for making the allowed-mentions object.
The message is already created in such "spec" consumer. So builder would be the imperative way to do so for the allowed mentions.

If you want to have this changed as "Spec" consumer or just as naming "Spec" let me know.

Reply:
Right, I'm saying the alternative would be `Message#setAllowedMentions(Consumer<AllowedMentionsSpec>)`. We have several existing nested specs already. Again, if we know we're going to do this style of Builder in the future anyways, I don't mind.